webrtc.html

```diff
@@ -12936,6 +12936,11 @@ <h2>Confidentiality of Communications</h2>
       <p>A mechanism, <code><a>peerIdentity</a></code>, is provided that gives
       Javascript the option of requesting media that the same javascript cannot
       access, but can only be sent to certain other entities.</p>
+      <p>Communication certificates may be opaquely shared with
+      <code>postMessage</code> in anticipation of future needs. User agents are
+      encouraged to not directly hold private keying material in these shared
+      objects, or limit sharing to the same origin, to reduce memory attack
+      surface.</p>
```

Comment:
If a memory-reading attack is possible, I'm not sure what an extra level of indirection gives you - pointers are made to be followed.
Won't an inconsistent limitation of sharing to the same origin mean that apps that want to share between origins will work in some browsers and not in others? Isn't that bad enough that we should either say MUST or drop the "limit sharing" language?

Should we mention that certificates can only be *used* in the origin context?

Reply:
> I'm not sure what an extra level of indirection gives you - pointers are made to be followed.

This is in reference to [existing language](http://w3c.github.io/webrtc-pc/#dom-rtccertificate-getfingerprints): *"In implementations where an RTCCertificate might not directly hold private keying material (it might be stored in a secure module), a reference to the private key can be held in the [[KeyingMaterial]] internal slot, allowing the private key to be stored and used."*

> Should we mention that certificates can only be used in the origin context?

We [already do](http://w3c.github.io/webrtc-pc/#h-note-31): *"the object cannot be used by any other origin than the one that originally created it."*

> Won't an inconsistent limitation of sharing to the same origin mean that apps that want to share between origins will work in some browsers and not in others? Isn't that bad enough that we should either say MUST or drop the "limit sharing" language?

I don't think it's that bad. You'd have to share it to a different origin and back again, basically. Who does that?

Yes I think we'd like to limit this normatively, but we're lacking a good way currently. I believe https://github.com/whatwg/html/issues/4939 was filed to fix that. Until then, we have this note. Happy to take suggestions.

Reply:
Clarified language to be about using a different process, and gave up trying to enforce sameorigin, based on https://github.com/w3c/webrtc-pc/issues/2343#issuecomment-549715469.